x-pack/plugins/uptime/public/components/monitor/synthetics/step_detail/waterfall/data_formatting.test.ts

```diff
@@ -456,4 +481,70 @@ describe('getSeriesAndDomain', () => {
       }
     `);
   });
+
+  it('handles formatting when mime type is not mapped to a specific mime type bucket', () => {
+    const actual = getSeriesAndDomain(networkItemsWithUncommonMimeType);
+    expect(actual).toMatchInlineSnapshot(`
```

Comment:
I would argue that this is larger than what we'd want for an inline snapshot. I'd say we should test behavior more specifically. 

What was the previous behavior? Was it just a missing color? If so, we could test for that etc. Maybe there need to be multiple assertions vs. the snapshot.

Reply:
To be more clear, this is past the size where my brain can easily process it, and it's not clear exactly what we're testing. My worry here is that a future change will come along and someone will quickly approve it without a close read.

Reply:
That makes sense to me, especially since this is testing a very specific part of the overall formatting logic.

Reply:
We did say we were ok using inline snaps to test API output, we probably should've had more specificity. I think an assertion that `series` contains an item with the expected MIME type, and a snap of the result from `find`ing it would be sufficient to cover the added logic.

Reply:
++, we did say that, however, this seems to not fit. We made that decision without real world examples, so, it's natural we'd find exceptions after putting it into practice. ++ to using `find` or something else.

To be clear, if we find that the principles we develop by don't serve us, it's always fine to challenge or re-evaluate them.